test/solidity/address/address.js

```diff
@@ -0,0 +1,146 @@
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const Constants = require('../../constants')
+const Utils = require('../../hts-precompile/utils')
+
+const TOP_UP_AMOUNT = ethers.utils.parseEther('1.0');
+const TRANSFER_AMOUNT = 1
+
+describe('Solidity Address tests:', function () {
+    let signers,contract, wallet, walletAddr, recipientContract, recipientAddr;
+
+    const tinybarToWeibar = (amount) => amount.mul(Utils.tinybarToWeibarCoef)
+    const weibarTotinybar = (amount) => amount.div(Utils.tinybarToWeibarCoef)
+
+    before(async function () {
+        signers = await ethers.getSigners()
+        wallet = signers[0];
+        walletAddr = await wallet.getAddress()
+
+        //deploy test contract
+        const factory = await ethers.getContractFactory(Constants.Contract.AddressContract)
+        contract = await factory.deploy()
+        await contract.deployed();
+
+        //deploy test contract
+        const calledFactory = await ethers.getContractFactory(Constants.Contract.Recipient)
+        recipientContract = await calledFactory.deploy()
+        await recipientContract.deployed();
+        recipientAddr = await recipientContract.address;
+
+        //top up the test contract with some funds
+        let tx = {
+            to: contract.address,
+            value: TOP_UP_AMOUNT
+        }
+        const topUpRes = await wallet.sendTransaction(tx)
+        topUpRes.wait();
+    })
+
+    it('<address>.balance', async function () {
+        const balance = await wallet.getBalance();
+        const res = await contract.getAddressBalance(walletAddr);
+        
+        expect(tinybarToWeibar(res)).to.equal(balance);
+        expect(tinybarToWeibar(res).gt(0)).to.be.true;
+    })
+
+    it('<address>.code', async function () {
```

Comment:
It would be nice to apply the best practice for naming - e.g the name of the it statement should be something like
it('should veirfy <address>.code works correctly')

Reply:
Done. Thank you :)